Automattic/wp-calypso
# Theme Tiers: Theme Upgrade Modal now is Personal Themes aware

Related to:
- https://github.com/Automattic/dotcom-forge/issues/4597
Builds on (it's the follow-up but no real Live Preview code is needed):
- https://github.com/Automattic/wp-calypso/pull/85648

## Proposed Changes

A few notes. This PR can't be merged directly. We don't have feature flags on the Live Preview so to test we need to have these changes un-feature flag aware. After testing that the Live Preview is working correctly, we are going to need to add a feature check in the `theme-upgrade-modal/index.tsx` so that the modal is gated. We can remove the Live Preview feature checks since they're just blocking the execution and the code isn't reachable unless you hack the URL and provide a Personal Theme, so it's safe to leave those checks removed.

* Added personal themes feature on Calypso.
* The Theme Upgrade Modal now correctly displays information regarding Personal Tier Themes.
* Live preview now displays the correct information when previewing Personal Tier Themes

## Testing Instructions



* Sandbox widgets.wp.com
* Upload the wpcom site editor artifact to your sandbox or build and sync [manually](https://github.com/Automattic/wp-calypso/blob/trunk/apps/wpcom-block-editor/README.md#build). 
* Open the Calyspo live link.
* Go to the LiTS and add ?flags=themes/tiers.
* Filter by Personal Themes and click on the details of any Personal Theme.
* On the Theme Details page, click on the Live Preview button.
* You should see the same elements, upsells, and CTAs as if you were previewing a Personal theme.
* You should see the upgrade modal when clicking on the CTA button.
* All CTAs and the upgrade modal should have the correct information.
* You can also test other places in our flows where the Upgrade Modal is being shown.
* The upgrade modal checkout flow should work as expected.

![Screenshot 2023-12-22 at 14 32 55](https://github.com/Automattic/wp-calypso/assets/1989914/3f5ff90d-806f-4f85-b2de-447d872cde6b)
![Screenshot 2023-12-22 at 14 32 34](https://github.com/Automattic/wp-calypso/assets/1989914/e1e85ea1-b7f5-494d-808b-7762c8acddeb)
![Screenshot 2023-12-22 at 14 32 27](https://github.com/Automattic/wp-calypso/assets/1989914/c32b4ba9-8e66-4355-b0c7-0b705bf0af40)




## Pre-merge Checklist
